### PR TITLE
Make migration more trustworthy and other bugfixes

### PR DESCRIPTION
**Pull request by Arachnid**
- Reconfigures it so only the owner of a name can authorise migration to a new contract if the current one has to be stopped.
 - Fixes a bug that allowed resetting the post-.eth-upgrade transfer address after it has been configured.
 - Fixes a bug that made it impossible to configure your domain after the emergency stop process has been invoked.
 - Fixes a bug that allowed new names to be registered after the emergency stop process has been invoked.
 - Fix a few lint errors

### CODE REVIEW

File: contracts/SubdomainRegistrar.sol [hunk 1]
```diff
@@ -135,16 +135,23 @@ contract SubdomainRegistrar is RegistrarInterface {
...
      */
-    function configureDomain(string name, uint price, uint referralFeePPM) {
+    function configureDomain(string name, uint price, uint referralFeePPM) public {
         configureDomainFor(name, price, referralFeePPM, msg.sender, 0x0);
     }
 
-    function configureDomainFor(string name, uint price, uint referralFeePPM, address owner, address transfer) public not_stopped owner_only(keccak256(name)) {
+    /**
+     * @dev Configures a domain, optionally transferring it to a new owner.
+
```
Comment (decanus) [inline contracts/SubdomainRegistrar.sol]: Remove eol, maybe add type comments.

File: contracts/SubdomainRegistrar.sol [hunk 2]
```diff
@@ -154,7 +161,12 @@ contract SubdomainRegistrar is RegistrarInterface {
 
         domain.price = price;
         domain.referralFeePPM = referralFeePPM;
-        domain.transferAddress = transfer;
+
+        if(domain.transferAddress != _transfer && _transfer != 0) {
+          domain.transferAddress = _transfer;
```
Comment (decanus) [inline contracts/SubdomainRegistrar.sol]: 4 space indent

File: contracts/SubdomainRegistrar.sol [hunk 3]
```diff
@@ -154,7 +161,12 @@ contract SubdomainRegistrar is RegistrarInterface {
 
         domain.price = price;
         domain.referralFeePPM = referralFeePPM;
-        domain.transferAddress = transfer;
+
+        if(domain.transferAddress != _transfer && _transfer != 0) {
```
Comment (decanus) [inline contracts/SubdomainRegistrar.sol]: Space after if

Comment (Arachnid) [reply contracts/SubdomainRegistrar.sol]: Bah.

**Comment by Arachnid** [comment]
PTAL.

**Pull request merged**